packages/node-resolve/src/index.js

```diff
@@ -250,4 +250,4 @@ export default function nodeResolve(opts = {}) {
       return idToPackageInfo.get(id);
     }
   };
-}
+};
```

Comment:
What you want to do is copy and freeze down here:

```suggestion
};

export const DEFAULTS = deepFreeze(deepMerge({}, defaults));
```

Using https://npm.runkit.com/deepmerge and https://www.npmjs.com/package/deep-freeze. We might also consider https://www.npmjs.com/package/fast-copy but it looks possibly unproven.

This may seem excessive, but believe me, you do not want users opening issues because an errant modification of defaults caused some obscure, hard to triage bug. These things in the bundler realm need to be locked down effectively.

Reply:
I don't see the point of deep-cloning here. Deep-freeze should be enough. Am I missing something?

Reply:
```js
const a = { x: { y: 1 } };
const b = deepFreeze(a);

b.x = 4;
console.log(a.x); // { y: 1 }
```

Reply:
https://github.com/rollup/plugins/pull/301#discussion_r409597038

> defaults should stay internal and we should be able to mutate it as needed internally

Reply:
Deep-freezing exports doesn't disallow mutating defaults, does it?

Reply:
We want to export a clone of `defaults` that is deeply frozen. We don't want the internal `defaults` to be frozen.